connector/profilingmetricsconnector/connector.go

```diff
@@ -159,26 +218,123 @@ func (c *profilesToMetricsConnector) collectFrameTypeCounts(dictionary pprofile.
 	}
 }
 
-// addAggregatedFrameTypeMetrics converts and adds frame type information as metric to the scopeMetrics.
-func (c *profilesToMetricsConnector) addAggregatedFrameTypeMetrics(origin origin, frameTypeCounts map[string]int64, scopeMetrics pmetric.ScopeMetrics, ts pcommon.Timestamp) {
-	for typ, count := range frameTypeCounts {
-		metric := scopeMetrics.Metrics().AppendEmpty()
-		metric.SetName(c.config.MetricsPrefix + "samples.frame_type." + typ)
-		metric.SetDescription("")
-		metric.SetUnit("1")
+// addMetrics converts and adds count information as metric to the scopeMetrics.
+func (c *profilesToMetricsConnector) addMetrics(origin origin, counts map[string]int64,
+	metricName, metricDesc, dataPointAttribute string,
+	scopeMetrics pmetric.ScopeMetrics, ts pcommon.Timestamp) {
+	if len(counts) == 0 {
+		return
+	}
 
-		sum := metric.SetEmptySum()
-		sum.SetIsMonotonic(true)
-		sum.SetAggregationTemporality(pmetric.AggregationTemporalityCumulative)
+	metric := scopeMetrics.Metrics().AppendEmpty()
+	metric.SetName(c.config.MetricsPrefix + metricName)
+	metric.SetDescription(metricDesc)
+	metric.SetUnit("1")
 
-		dataPoint := sum.DataPoints().AppendEmpty()
+	gauge := metric.SetEmptyGauge()
+	for typ, count := range counts {
+		dataPoint := gauge.DataPoints().AppendEmpty()
 		dataPoint.SetIntValue(count)
 		dataPoint.SetTimestamp(ts)
-
+		dataPoint.Attributes().PutStr(dataPointAttribute, typ)
 		dataPoint.Attributes().PutStr("profile.type_unit", origin.typ+"_"+origin.unit)
 	}
 }
 
+// collectClassificationCounts walks all locations/frames of a sample and collects the classification information.
+func (c *profilesToMetricsConnector) collectClassificationCounts(dictionary pprofile.ProfilesDictionary, locationIndices pcommon.Int32Slice, sample pprofile.Sample, classificationCounts map[string]int64) {
```

Comment:
If we  use a `map[string]int64` for `classificationCounts` we're dropping the frame type information, meaning we are mixing packages from Java and Go in the same map. 

If we instead use a `map[string]map[string]int64` we can keep frame type around which will later allow us to easily split the counts by frame type for visualizations, for example by adding a frametype attribute to each metric.